app/controllers/katello/api/v2/environments_controller.rb

```diff
@@ -44,15 +44,17 @@ class Api::V2::EnvironmentsController < Api::V2::ApiController
 
     api :GET, "/environments", N_("List environments in an organization")
     api :GET, "/organizations/:organization_id/environments", N_("List environments in an organization")
-    param :organization_id, :number, :desc => N_("organization identifier"), :required => true
+    param :organization_id, :number, :desc => N_("organization identifier")
     param :library, [true, false], :desc => N_("set true if you want to see only library environments")
     param :name, String, :desc => N_("filter only environments containing this name")
     def index
       respond(:collection => scoped_search(index_relation.uniq, :name, :desc, :resource_class => KTEnvironment))
     end
 
     def index_relation
-      query = KTEnvironment.readable.where(:organization_id => @organization.id)
+      query = KTEnvironment.readable
+      find_organization if params[:organization_id]
```

Comment:
You could avoid this by adding `:index` to the list for the `find_optional_organization` filter at the top